### PR TITLE
Dashboard links: Places drop down list so it's always visible

### LINKED ISSUE #27900: Link menu dropdown items get hidden under the left-side menu
**What happened**:
Then link menu dropdown items get hidden under the left-side menu so they are not fully visible to the user.
LINKS_SUBMENU_HIDDEN_UNDER_LEFT_MENU

**What you expected to happen**:
The link menu dropdown items are fully visible to the user. 

**How to reproduce it (as minimally and precisely as possible)**:
- choose an arbitrary panel and add a number of template variables so that they occupy around 60 to 80% of the screen width
- add a link menu to that panel and make sure the link menu items wrap to the next line (as displayed in the screenshot attached)
- make sure the last element of the link menu (the one that wraps to the next line) is a dropdown and add several menu items to that dropdown
- click on that dropdown menu to visualize its menu items - they appear hidden under the left-side menu

**Anything else we need to know?**:
It's possibly a question of applying a `pull-left` on the wrapped menu items instead of `pull-right` that is currently applied; this is what makes them hide under the left-side menu.

I'm attaching a version where `pull-left` is applied to the `ul` element making its menu items (li elements) display just right.
PULL_LEFT_APPLIED_TO_THE_MENU

**Environment**:
- Grafana version: 7.1.3
- Data source type & version: N/A
- OS Grafana is installed on: macOS Catalina 10.15.7
- User OS & Browser: macOS Catalina 10.15.7, Chrome Version 85.0.4183.121
- Grafana plugins: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by maknik**
**What this PR does / why we need it**:
Calculating whether to place the list on the right or left edge of the parent. List can't be always pull-right or pull-left, because if dashboard link is to close to the left or right side of the container, part of list may be hidden.

**Which issue(s) this PR fixes**:

Fixes #27900 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by maknik** [comment]
@hugohaggmark Unfortunately I had problem with contributing license on my previous PR

**Review by hugohaggmark — APPROVED** [review]
Thank you again for contributing @maknik ⭐

**Comment by hugohaggmark** [comment]
@torkelo should I've used `backport 7.3.x` label here instead?

**Comment by torkelo** [comment]
@hugohaggmark  Don't think it will work on this PR as it is from a fork, was a limitation that many encountered with github actions even on merged PRs do not have access to repo secrets. I found no solution only many others had run into the same issue. But I did more research and found a solution (I hope) https://github.com/grafana/grafana/pull/28352

**Comment by hugohaggmark** [comment]
> @hugohaggmark Don't think it will work on this PR as it is from a fork, was a limitation that many encountered with github actions even on merged PRs do not have access to repo secrets. I found no solution only many others had run into the same issue. But I did more research and found a solution (I hope) #28352

Cool, thank you!

**Comment by kylebrandt** [comment]
Removed cp label because it is in beta2, and see this in the v7.3.x branch already

File: public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx [hunk 1]
```diff
@@ -79,6 +81,16 @@ export class DashboardLinksDashboard extends PureComponent<Props, State> {
...
   };
 
+  getDropdownLocationCssClass = (): string => {
+    const [pullLeftCssClass, pullRightCssClass] = ['pull-left', 'pull-right'];
+    const wrapper = this.wrapperRef.current;
+    const list = this.listItemRef.current;
+    if (!wrapper || !list) {
+      return pullRightCssClass;
+    }
+    return wrapper.offsetLeft > list.offsetWidth - wrapper.offsetWidth ? pullRightCssClass : pullLeftCssClass;
```
Comment (torkelo) [inline public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx:91]: this logic does not make sense to me, should not the viewport width be part of this calculation?

**Pull request merged**